File: typescript/pay.ts
/**
 * TypeScript + Rosud — Payment between AI agents
 */
import Rosud from 'rosud';

const client = new Rosud({ apiKey: process.env.ROSUD_API_KEY! });

async function main() {
  // Check balance
  const balance = await client.wallets.balance();
  console.log(`Balance: ${balance.usdc} USDC`);

  // Send payment
  const payment = await client.payments.create({
    amount: 0.10,
    to: '0x742d35Cc6634C0532925a3b8D4C9E3Ff9C4A6bB',
    memo: 'inference_fee',
  });
  console.log(`Payment: ${payment.status} | ${payment.txHash}`);
}

main().catch(console.error);

File: basic/pay.py
"""
Basic Rosud example — single payment between agents.
"""
import rosud

client = rosud.Rosud(api_key="rosud_live_xxx")
# or: export ROSUD_API_KEY=rosud_live_xxx

# 1. Check balance
balance = client.wallets.balance()
print(f"Balance: {balance.usdc} USDC")

# 2. Send payment
payment = client.payments.create(
    amount=0.10,
    to="0x742d35Cc6634C0532925a3b8D4C9E3Ff9C4A6bB",
    memo="research_task_fee"
)
print(f"Payment: {payment.status} | tx: {payment.tx_hash}")

# 3. Verify
result = client.payments.get(payment.id)
print(f"Confirmed: {result.confirmed_at}")
